src/main/java/com/epam/vsharstuk/springboot_mvc/dao/CarRepositoryImpl.java

```diff
@@ -0,0 +1,145 @@
+package com.epam.vsharstuk.springboot_mvc.dao;
+
+import com.epam.vsharstuk.springboot_mvc.model.Car;
+import com.epam.vsharstuk.springboot_mvc.service.CarSearchCriteria;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Repository
+public class CarRepositoryImpl implements  CarRepository{
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
```

Comment:
Spring provides convenient class `NamedJdbcTemplate`. It allows to bind query parameters by name.